tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_vector_sum_of_squares_kk.hpp

```diff
@@ -2,21 +2,66 @@
 #ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL_P1673_BITS_KOKKOSKERNELS_VEC_SUM_OF_SQUARES_HPP_
 #define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL_P1673_BITS_KOKKOSKERNELS_VEC_SUM_OF_SQUARES_HPP_
 
+#include "signal_kokkos_impl_called.hpp"
+
 namespace KokkosKernelsSTD {
 
 template<class ExecSpace,
          class ElementType,
          std::experimental::extents<>::size_type ext0,
          class Layout,
-         class Accessor,
          class Scalar>
 std::experimental::linalg::sum_of_squares_result<Scalar>
-vector_sum_of_squares(kokkos_exec<ExecSpace>,
-		      std::experimental::mdspan<ElementType, std::experimental::extents<ext0>, Layout, Accessor> v,
+vector_sum_of_squares(kokkos_exec<ExecSpace> /*kexe*/,
+		      std::experimental::mdspan<
+		      ElementType,
+		      std::experimental::extents<ext0>,
+		      Layout,
+		      std::experimental::default_accessor<ElementType>> x,
 		      std::experimental::linalg::sum_of_squares_result<Scalar> init)
 {
-  std::cout << "vector_sum_of_squares: kkernels impl missing !!!\n";
-  return init;
+
+  Impl::signal_kokkos_impl_called("vector_sum_of_squares");
+
+  auto x_view = Impl::mdspan_to_view(x);
+  std::experimental::linalg::sum_of_squares_result<Scalar> result;
+
+  // FRIZZI:
+  // (1) this works only for types that have a reduction identity.
```

Comment:
Just use `T{}` for the reduction identity.  We may need some additional wording in P1673 for the parallel algorithms.  I've filed a wording issue for that: https://github.com/kokkos/stdBLAS/issues/188